packages/types/src/checksum.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * An object that provides a checksum of data provided in chunks to `update`.
+ * The checksum may be performed incrementally as chunks are received or all
+ * at once when the checksum is finalized, depending on the underlying
+ * implementation.
+ *
+ * It's recommended to compute checksum incrementally to avoid reading the
+ * entire payload in memory.
+ *
+ * The constructor may accept an optional secret key while computing checksum
+ * value, when using HMAC. If provided, this secret key would be used when
+ * computing checksum.
```

Comment:
Maybe document it for classes that implement this interface

```suggestion
 *
 * A class that implements this interface may accept an optional secret key in its
 * constructor while computing checksum value, when using HMAC. If provided,
 * this secret key would be used when computing checksum.
```